binary_search_tree/tree.py

```diff
@@ -16,44 +16,124 @@ def __init__(self):
 
     # Time Complexity: 
     # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    def add_helper (self, current, key, value):
+        if current == None:
+            return TreeNode(key, value)
+        elif current.key >= key:
+            current.left = self.add_helper(current.left, key, value)
+        else: 
+            current.right = self.add_helper(current.right, key, value)
+        return current 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def add(self, key, value=None):
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
+    def find_helper(self, current, key):
```

Comment:
🤓 Style note: We like to add a blank line between the end of one function and the beginning of another